### PR TITLE
Structure_Engine: Update to area loads visualisation

### LINKED ISSUE #1965: Structure_Engine: Visualising loads in local axis needs update with local orientation
#### Description:

After change of local orientation for panels, visualisation of loads need to be aligned.

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### Issues addressed by this PR

Closes #1965 

### Test files

https://burohappold.sharepoint.com/:f:/s/BHoM/Em_LdUZqgX1KuIVqdNMiGJcBlPpuLhdHOREUu3lv_eSfTQ?e=ARRPJB

Test files with updated visualisations:

https://burohappold.sharepoint.com/:f:/s/BHoM/EhgghoopPXpHispXwTlB240BQpShlJn1oWapcJUVkzDSUA?e=uSOKbG

### Changelog

- Area load visualisation to make use of LocalOrientation for displaying local loads
- Add support for displaying Arealoads on FEMeshes
- Add support for visualisation of ContourLoad and GeometricLineLoad

### CODE REVIEW

**Review by pawelbaran — APPROVED** [review]
LGTM, could only complain about the distribution of the blank lines 🤣

File: Structure_Engine/Query/PointGrid.cs [hunk 1]
```diff
@@ -113,6 +113,35 @@ public static List<Point> PointGrid(this Panel panel)
...
+        [Output("grid", "Rectangular grid of points on the FEMesh.")]
+        public static List<List<Point>> PointGrid(this FEMesh mesh)
+        {
+            return mesh.Faces.Select(x => x.PointGrid(mesh)).ToList();
+        }
+
+        [Description("Generates a rectangular grid of points on the each face of the FEMesh. Used for load visualisation.")]
+        [Input("mesh", "The FEMesh to generate a grid on.")]
+        [Output("grid", "Rectangular grid of points on the FEMesh.")]
+        public static List<Point> PointGrid(this FEMeshFace face, FEMesh mesh)
```
Comment (pawelbaran) [inline Structure_Engine/Query/PointGrid.cs:129]: face description missing

File: Structure_Engine/Query/PointGrid.cs [hunk 2]
```diff
@@ -113,6 +113,35 @@ public static List<Point> PointGrid(this Panel panel)
...
+
+        [Description("Generates a rectangular grid of points on the each face of the FEMesh. Used for load visualisation.")]
+        [Input("mesh", "The FEMesh to generate a grid on.")]
+        [Output("grid", "Rectangular grid of points on the FEMesh.")]
+        public static List<List<Point>> PointGrid(this FEMesh mesh)
+        {
+            return mesh.Faces.Select(x => x.PointGrid(mesh)).ToList();
+        }
+
+        [Description("Generates a rectangular grid of points on the each face of the FEMesh. Used for load visualisation.")]
```
Comment (pawelbaran) [inline Structure_Engine/Query/PointGrid.cs:126]: not each face but one only?

File: Structure_Engine/Query/Visualize.cs [hunk 3]
```diff
@@ -462,6 +478,127 @@ public static List<ICurve> Visualize(this PointVelocity pointVelocity, double sc
             return arrows;
         }
 
+        /***************************************************/
+
+        [Description("Draws arrows representing the load along the edges of the contour of the load.")]
+        [Input("pointVelocity", "The node load to visualise.")]
```
Comment (pawelbaran) [inline Structure_Engine/Query/Visualize.cs:484]: contourLoad it is

File: Structure_Engine/Query/Visualize.cs [hunk 4]
```diff
@@ -462,6 +478,127 @@ public static List<ICurve> Visualize(this PointVelocity pointVelocity, double sc
...
+            arrows.AddRange(ConnectedArrows(contourLoad.Contour.SubParts(), forceVec, asResultants, orientation, 1, true));
+
+
+            return arrows;
+        }
+
+        /***************************************************/
+
+        [Description("Draws arrows representing the load along the length of the line of the load.")]
+        [Input("pointVelocity", "The node load to visualise.")]
```
Comment (pawelbaran) [inline Structure_Engine/Query/Visualize.cs:530]: line load it is

**Review by pawelbaran — CHANGES_REQUESTED** [review]
Although descriptions require another look.

**Comment by IsakNaslundBh** [comment]
Thanks @pawelbaran . Think I have addressed your points (also deleted some blank lines :) )

**Review by pawelbaran — APPROVED** [review]
LGTM now 👍

**Pull request merged**